test/functional/cinder_controller_test.go

```diff
@@ -863,6 +863,75 @@ var _ = Describe("Cinder controller", func() {
 
 	})
 
+	When("Cinder CR instance is built with ExtraMounts", func() {
+		BeforeEach(func() {
+			rawSpec := map[string]interface{}{
+				"secret":              SecretName,
+				"databaseInstance":    "openstack",
+				"rabbitMqClusterName": "rabbitmq",
+				"extraMounts":         GetExtraMounts(),
+				"cinderAPI": map[string]interface{}{
+					"containerImage": cinderv1.CinderAPIContainerImage,
+				},
+				"cinderScheduler": map[string]interface{}{
+					"containerImage": cinderv1.CinderSchedulerContainerImage,
+				},
+				"cinderVolumes": map[string]interface{}{
+					"volume1": map[string]interface{}{
+						"containerImage": cinderv1.CinderVolumeContainerImage,
+					},
+				},
+			}
+
+			DeferCleanup(th.DeleteInstance, CreateCinder(cinderTest.Instance, rawSpec))
+			DeferCleanup(k8sClient.Delete, ctx, CreateCinderMessageBusSecret(cinderTest.Instance.Namespace, cinderTest.RabbitmqSecretName))
+			DeferCleanup(
+				mariadb.DeleteDBService,
+				mariadb.CreateDBService(
+					cinderTest.Instance.Namespace,
+					GetCinder(cinderTest.Instance).Spec.DatabaseInstance,
+					corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{{Port: 3306}},
+					},
+				),
+			)
+			infra.SimulateTransportURLReady(cinderTest.CinderTransportURL)
+			DeferCleanup(infra.DeleteMemcached, infra.CreateMemcached(namespace, cinderTest.MemcachedInstance, memcachedSpec))
+			infra.SimulateMemcachedReady(cinderTest.CinderMemcached)
+			keystoneAPIName := keystone.CreateKeystoneAPI(cinderTest.Instance.Namespace)
+			DeferCleanup(keystone.DeleteKeystoneAPI, keystoneAPIName)
+			mariadb.SimulateMariaDBAccountCompleted(cinderTest.Database)
+			mariadb.SimulateMariaDBDatabaseCompleted(cinderTest.Database)
+			th.SimulateJobSuccess(cinderTest.CinderDBSync)
+			keystone.SimulateKeystoneEndpointReady(cinderTest.CinderKeystoneEndpoint)
+		})
+
+		It("Check the extraMounts of the resulting StatefulSets", func() {
+			th.SimulateStatefulSetReplicaReady(cinderTest.CinderAPI)
+			th.SimulateStatefulSetReplicaReady(cinderTest.CinderScheduler)
+			// Retrieve the generated resources
+			volume := cinderTest.CinderVolumes[0]
+			th.SimulateStatefulSetReplicaReady(volume)
+			ss := th.GetStatefulSet(volume)
+			// Check the resulting deployment fields
+			Expect(int(*ss.Spec.Replicas)).To(Equal(1))
+			Expect(ss.Spec.Template.Spec.Volumes).To(HaveLen(15))
```

Comment:
could also use https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/test/helpers/volumes.go#L27 . to check that the volume with the expected name exist.

Reply:
Yes, I totally agree to this. Thank you, I wasn't aware about lib-common providing such assertion, and I'm going to use it. In general I don't like much  checking the `Volumes` (or `VolumeMount`) array length by hardcoding a value, and the assertion solves this problem providing a more solid check!